tools/ExampleTester/GeneratedExample.cs

```diff
@@ -127,46 +131,52 @@ bool ValidateOutput()
                 ? new object[] { Metadata.ExecutionArgs ?? new string[0] }
                 : new object[0];
 
-            var oldOut = Console.Out;
             List<string> actualLines;
             Exception? actualException = null;
-            try
+            lock (ConsoleAccessLock)
```

Comment:
With this lock in place, do we effectively have "parallel build, sequential execution"? That sounds fine to me - I would imagine the build is the main timing bottleneck.